To: vim_dev@googlegroups.com
Subject: Patch 7.3.274
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.274
Problem:    With concealed characters tabs do not have the right size.
Solution:   Use VCOL_HLC instead of vcol. (Eiichi Sato)
Files:	    src/screen.c


*** ../vim-7.3.273/src/screen.c	2011-03-22 13:29:20.000000000 +0100
--- src/screen.c	2011-08-10 14:22:47.000000000 +0200
***************
*** 4252,4258 ****
  		{
  		    /* tab amount depends on current column */
  		    n_extra = (int)wp->w_buffer->b_p_ts
! 				   - vcol % (int)wp->w_buffer->b_p_ts - 1;
  #ifdef FEAT_MBYTE
  		    mb_utf8 = FALSE;	/* don't draw as UTF-8 */
  #endif
--- 4252,4258 ----
  		{
  		    /* tab amount depends on current column */
  		    n_extra = (int)wp->w_buffer->b_p_ts
! 				   - VCOL_HLC % (int)wp->w_buffer->b_p_ts - 1;
  #ifdef FEAT_MBYTE
  		    mb_utf8 = FALSE;	/* don't draw as UTF-8 */
  #endif
*** ../vim-7.3.273/src/version.c	2011-08-10 13:21:30.000000000 +0200
--- src/version.c	2011-08-10 14:23:38.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     274,
  /**/

-- 
   Arthur pulls Pin out.  The MONK blesses the grenade as ...
ARTHUR:  (quietly) One, two, five ...
GALAHAD: Three, sir!
ARTHUR:  Three.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
